tests/unit/test_platforms_get_platform.py

```diff
@@ -0,0 +1,199 @@
+        (
+            {
+                'remote': {'host': 'localhost'},
+                'job': {
+                    'batch system': None,
+                    'batch submit command template': None,
+                    'execution polling intervals': None
+                }
+            },
+            'localhost'
+        ),
+        (
+            {
+                'remote': {'host': 'cylcdevbox'},
+                'job': {
+                    'batch system': None,
+                    'batch submit command template': None,
+                    'execution polling intervals': None
+                }
+            },
+            'cylcdevbox'
+        )
+    ]
+)
+def test_get_platform_using_platform_from_job_info(
+    mock_glbl_cfg, task_conf, expected_platform_name
+):
+    """Calculate platform from Cylc 7 config: n.b. If this fails we don't
+    warn because this might lead to many thousands of warnings
+
+    This should not contain a comprehensive set of use-cases - these should
+    be coverend by the unit tests for `platform_from_host_items`
+    """
+    mock_glbl_cfg(
+        'cylc.flow.platforms.glbl_cfg',
+        '''
+        [platforms]
+            [[ras_el_hanout]]
+                hosts = rose, chilli, cumin, paprika
+                batch system = slurm
+            [[spice_bg]]
+                hosts = rose, chilli, cumin, paprika
+            [[local_batch_system]]
+                hosts = localhost
+                batch system = batchyMcBatchFace
+            [[cylcdevbox]]
+                hosts = cylcdevbox
+        '''
+    )
+    assert get_platform(task_conf)['name'] == expected_platform_name
+
+
+def test_get_platform_warn_mode(caplog):
+    task_conf = {
+        'remote': {'host': 'cylcdevbox'},
+        'job': {
+            'batch system': 'pbs',
+            'batch submit command template': 'some template'
+        }
+    }
+    output = get_platform(task_conf, warn_only=True)
+    for forbidden_item in (
+        'batch submit command template = some template',
+        'host = cylcdevbox',
+        'batch system = pbs'
+    ):
+        assert forbidden_item in output
+
+
+def test_get_platform_groups_basic(mock_glbl_cfg):
+    # get platform from group works.
+    mock_glbl_cfg(
+        'cylc.flow.platforms.glbl_cfg',
+        '''
+        [platforms]
+            [[aleph]]
+                hosts = aleph
+            [[bet]]
+                hosts = bet
+
+        [platform groups]
+            [[hebrew_letters]]
+                platforms = aleph, bet
+        '''
+    )
+    output = get_platform('hebrew_letters')
+    assert output['group'] == 'hebrew_letters'
+    names = []
+    # this number keeps the test fast but give about 1.6x10^-30 chance of
+    # a false negative.
+    flakiness_n = 100
+    flakiness = 2 / 2 ** flakiness_n
+    for i in range(flakiness_n):
+        names.append(get_platform('hebrew_letters')['name'])
+        if 'aleph' in names and 'bet' in names:
+            break
```

Comment:
With this test written in Python there might be a nicer solution, the Python random library is only psudo-random. You can control the series of random numbers it produces by setting the [random seed](https://docs.python.org/3/library/random.html#random.seed).

```python
random.seed(1234)
assert [
    get_platform('hebrew_letters')['name']
    for _ in range(N)
] == [
    # results should come out in the same order each time
]
```

Reply:
This will, I think work rather well. Thanks for the point!